ocaml/parsing/ast_invariants.ml

```diff
@@ -107,6 +125,7 @@ let iterator =
     | Pexp_new id -> simple_longident id
     | Pexp_record (fields, _) ->
       List.iter (fun (id, _) -> simple_longident id) fields
+    | Pexp_function _ | Pexp_fun _ -> non_jane_syntax_function loc
```

Comment:
I think it's a good idea to try to get this PR merged before all ppxes are migrated to the new system. That means I will need to:
  (a) remove this check for ast_invariants
  (b) ensure that existing ppxes build with this PR.

I'm leaving this comment unresolved as a proxy for both changes, though it's only attached to (a).